### PR TITLE
Generate build & test matrix from script

### PR DESCRIPTION
**Pull request by sunmou99**
### Description
Generate build & test matrix from script:
build matrix format:   `unity_version-build_os-platform-ios_sdk`
test matrix format:   `unity_version-build_os-platform-test_device-test_os`

Updated some configs.

***
### Testing
Build & Test against one testapp: https://github.com/firebase/firebase-unity-sdk/actions/runs/3276829085

Also, see comment below
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [x] Other, such as a build process or documentation change.
***

### CODE REVIEW

File: scripts/gha/print_matrix_configuration.py [hunk 1]
```diff
@@ -80,82 +92,82 @@
...
+    LINUX: {
       "version": "2019.4.40f1",
-      "packages": {"Default": ["Unity"], "Android": ["Android"], "iOS": ["Ios"], "tvOS": ["appletv"], "Windows": ["Windows-mono"], "macOS": ["Mac-mono"], "Linux": None}
+      "packages": {"Default": ["Unity"], ANDROID: ["Android"], IOS: ["Ios"], TVOS: ["appletv"], WINDOWS: ["Windows-mono"], MACOS: ["Mac-mono"], LINUX: None}
     }
   },
   "2018": {
-    _WINDOWS: {
+    WINDOWS: {
       "version": "2018.4.36f1",
```
Comment (cynthiajoan) [inline scripts/gha/print_matrix_configuration.py:128]: we can remove 2018 support now :)

File: scripts/gha/print_matrix_configuration.py [hunk 2]
```diff
@@ -262,6 +249,114 @@ def print_value(value, config_parms_only=False):
     print(json.dumps(value))
 
 
+def get_testapp_build_matrix(matrix_type, unity_versions, platforms, build_os, ios_sdk):
+  # matrix structure:
+  # {
+  #   "unity_version":"unity_version",
+  #   "platform":"platform",
+  #   "os":"os",
```
Comment (DellaBitta) [inline scripts/gha/print_matrix_configuration.py:257]: Should this be `"os":"ios",`  ?

File: scripts/gha/print_matrix_configuration.py [hunk 3]
```diff
@@ -262,6 +249,114 @@ def print_value(value, config_parms_only=False):
...
+  if matrix_type: unity_versions = get_value("integration_tests", matrix_type, "unity_versions")
+  if matrix_type: platforms = filter_build_platforms(get_value("integration_tests", matrix_type, "platforms"))
+  else: platforms = filter_build_platforms(platforms)
+  if matrix_type: build_os = get_value("integration_tests", matrix_type, "build_os")
+  if matrix_type: ios_sdk = get_value("integration_tests", matrix_type, "mobile_test_on")
+
+  # generate base matrix: combinations of (unity_versions, platforms, build_os)
+  li = list(itertools.product(unity_versions, platforms, build_os))
+  matrix = {"include": []}
+  for l in li:
```
Comment (DellaBitta) [inline scripts/gha/print_matrix_configuration.py:270]: Could you expand these variable names? For me I would think `li` would be list item and `l` woudl be list, but it's the opposite and a bit confusing.

File: scripts/gha/print_matrix_configuration.py [hunk 4]
```diff
@@ -262,6 +249,114 @@ def print_value(value, config_parms_only=False):
...
+  # {
+  #   "unity_version":"unity_version",
+  #   "os":"os",
+  # }
+
+  if PLAYMODE not in platforms: return ""
+  if matrix_type: unity_versions = get_value("integration_tests", matrix_type, "unity_versions")
+  if matrix_type: build_os = get_value("integration_tests", matrix_type, "build_os")
+
+  li = list(itertools.product(unity_versions, build_os))
```
Comment (DellaBitta) [inline scripts/gha/print_matrix_configuration.py:297]: here, too.

File: scripts/gha/print_matrix_configuration.py [hunk 5]
```diff
@@ -262,6 +249,114 @@ def print_value(value, config_parms_only=False):
...
+  matrix = {"include": []}
+  for l in li:
```
Comment (DellaBitta) [inline scripts/gha/print_matrix_configuration.py:299]: and here.

File: scripts/gha/print_matrix_configuration.py [hunk 6]
```diff
@@ -262,6 +249,114 @@ def print_value(value, config_parms_only=False):
...
+
+  if matrix_type: unity_versions = get_value("integration_tests", matrix_type, "unity_versions")
+  if matrix_type: platforms = get_value("integration_tests", matrix_type, "platforms")
+  if matrix_type: build_os = get_value("integration_tests", matrix_type, "build_os")
+  if matrix_type: mobile_device_types = get_value("integration_tests", matrix_type, "mobile_test_on")
+
+  # generate base matrix: combinations of (unity_versions, platforms, build_os)
+  li = list(itertools.product(unity_versions, platforms, build_os))
+  matrix = {"include": []}
+  for l in li:
```
Comment (DellaBitta) [inline scripts/gha/print_matrix_configuration.py:327]: ^

**Review by DellaBitta — APPROVED** [review]

**Review by cynthiajoan — APPROVED** [review]

**Pull request merged**